### PR TITLE
Implementation

### PR DESCRIPTION
**Pull request by markGilchrist**
Currently this is tested for XCCase, I have left the functions that use nimble as internal because they will cause a function name clash at comile time.

 I feel would should consider merging this to develop then splitting this into a two target project so that users can import either a `RxTestExt` or a `RxTestExtNimble`.

we could add a prefix to all the nimble function name which would prevent a function name clash. 

Please let me know your thoughts

### CODE REVIEW

**Comment by sindhu238** [comment]
When we split nimble and xctest code to two targets, we can have same
function names for both of them as long as they both are not in the same
target.

**Comment by markGilchrist** [comment]
@sindhu238 Yes Then users could import one or the other, Is this correct @andecoder ?

File: RxTestExt/Classes/TestableObserverExtensions.swift [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+    var errors: [Error] {
+        return events.compactMap { $0.value.error }
+    }
+
+    var errorCount: Int {
+        return errors.count
+    }
+
+    var completeions: Int {
```
Comment (sindhu238) [inline RxTestExt/Classes/TestableObserverExtensions.swift]: Typo

**Review by sindhu238 — APPROVED** [review]

**Comment by andecoder** [comment]
> @sindhu238 Yes Then users could import one or the other, Is this correct @andecoder ?

I believe so

**Pull request merged**